enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/roles/Follower.java

```diff
@@ -167,49 +192,127 @@ public Outcome handle( RaftMessages.NewEntry.BatchRequest batchRequest ) throws
     {
+        Outcome handle( RaftMessages.PreVote.Response response, Outcome outcome, ReadableRaftState ctx, Log log ) throws IOException;
+    }
 
-        PreVoteSupportedHandler( ReadableRaftState ctx, Log log )
+    private static class PreVoteSupportedHandler implements ElectionTimeoutHandler
+    {
+        public Outcome handle( RaftMessages.Timeout.Election election, Outcome outcome, ReadableRaftState ctx, Log log ) throws IOException
         {
-            super( ctx, log );
+            log.info( "Election timeout triggered" );
+            if ( Election.startPreElection( ctx, outcome, log ) )
+            {
+                outcome.setPreElection( true );
+            }
+            return outcome;
         }
 
+        private static ElectionTimeoutHandler instance = new PreVoteSupportedHandler();
+    }
+
+    private static class PreVoteUnsupportedHandler implements ElectionTimeoutHandler
+    {
         @Override
-        public Outcome handle( RaftMessages.Timeout.Election election ) throws IOException
+        public Outcome handle( RaftMessages.Timeout.Election election, Outcome outcome, ReadableRaftState ctx, Log log ) throws IOException
         {
             log.info( "Election timeout triggered" );
-            if ( Election.startPreElection( ctx, outcome, log ) )
+            if ( Election.startRealElection( ctx, outcome, log ) )
             {
-                outcome.setPreElection( true );
+                outcome.setNextRole( CANDIDATE );
+                log.info( "Moving to CANDIDATE state after successfully starting election" );
             }
             return outcome;
         }
+
+        private static ElectionTimeoutHandler instance = new PreVoteUnsupportedHandler();
     }
 
-    class PreVoteActiveHandler extends PreVoteSupportedHandler
+    private static class PreVoteUnsupportedRefusesToLead implements ElectionTimeoutHandler
     {
+        @Override
+        public Outcome handle( RaftMessages.Timeout.Election election, Outcome outcome, ReadableRaftState ctx, Log log ) throws IOException
+        {
+            log.info( "Election timeout triggered but refusing to be leader" );
+            return outcome;
+        }
+
+        private static ElectionTimeoutHandler instance = new PreVoteUnsupportedRefusesToLead();
+    }
 
-        PreVoteActiveHandler( ReadableRaftState ctx, Log log )
+    private static class PreVoteSupportedRefusesToLeadHandler implements ElectionTimeoutHandler
+    {
+        @Override
+        public Outcome handle( RaftMessages.Timeout.Election election, Outcome outcome, ReadableRaftState ctx, Log log ) throws IOException
         {
-            super( ctx, log );
+            log.info( "Election timeout triggered but refusing to be leader" );
+            Set<MemberId> memberIds = ctx.votingMembers();
+            if ( memberIds != null && memberIds.contains( ctx.myself() ) )
+            {
+                outcome.setPreElection( true );
```

Comment:
`preElection` needs to be true to participate in pre elections (`PreVoteRequestVotingHandler`). However it won't send pre vote requests and it will ignore pre vote responses.